test/api/SettingTest.ts

```diff
@@ -21,6 +22,98 @@ chai.use(responseHelper);
 const testData: TestData = new TestData();
 let initialTenant: Tenant;
 
+// Update pricing setting to have sensitive data to test on it
+async function updatePricingWithSensitiveData():Promise<any> {
+  const read = await testData.centralService.settingApi.readAll({ 'Identifier': 'refund' }, {
+    limit: TestConstants.UNLIMITED,
+    skip: 0
+  });
+  expect(read.status).to.equal(200);
+  expect(read.data.count).to.equal(1);
+  const clientSecret = 'a8242e0ed0fa70aee7c802e41e1c7c3b';
+  testData.data = JSON.parse(`{
+      "id":"${read.data.result[0].id}",
+      "identifier":"refund",
+      "sensitiveData":["content.concur.clientSecret"],
+      "content":{
+        "type" : "concur",
+        "concur" : {
+            "authenticationUrl" : "https://url.com",
+            "apiUrl" : "https://url.com",
+            "appUrl" : "https://url.com",
+            "clientId" : "6cf707fb-9161-48fa-94fe-9e003be680df",
+            "clientSecret" : "${clientSecret}",
+            "paymentTypeId" : "gWtUBRXx$s3h0bNdgyv9gwiHLnCGMF",
+            "expenseTypeCode" : "01104",
+            "policyId" : "1119",
+            "reportName" : "E-Car Charging"
+        }
+    }
+  }`);
+  const update = await testData.centralService.updateEntity(testData.centralService.settingApi, testData.data);
+  expect(update.status).to.equal(200);
+}
+
+function getCryptoTestData(originalCryptoObject, newKeyProperties:CryptoKeyProperties) {
+  return JSON.parse(`{
+    "id":"${originalCryptoObject.id}",
+    "identifier":"${originalCryptoObject.identifier}",
+    "sensitiveData":[],
+    "content":{
+      "type":"crypto",
+      "crypto" : {
+        "key" : "${Utils.generateRandomKey(newKeyProperties)}",
+        "keyProperties" : {
+            "blockCypher" : "${newKeyProperties.blockCypher}",
+            "blockSize" : "${newKeyProperties.blockSize}",
+            "operationMode" : "${newKeyProperties.operationMode}"
+        },
+        "formerKey" : "${originalCryptoObject.content.crypto.key}",
+        "formerKeyProperties" : {
+            "blockCypher" : "${originalCryptoObject.content.crypto.keyProperties.blockCypher}",
+            "blockSize" : "${originalCryptoObject.content.crypto.keyProperties.blockSize}",
+            "operationMode" : "${originalCryptoObject.content.crypto.keyProperties.operationMode}"
+        },
+        "migrationToBeDone" : true
+      }
+    }
+}`);
+}
+
+async function getCurrentCryptoData() {
```

Comment:
All following methods do more than just get, reset or update, they are also doing some validation, checks of the outputs. Their names shall reflect it.